hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/hive/HoodieCombineHiveInputFormat.java

```diff
@@ -562,8 +568,35 @@ public RecordReader getRecordReader(InputSplit split, JobConf job, Reporter repo
     }
   }
 
+  private void invokePushProjectAndFilters(JobConf job, Class<?> inputFormatClass, Path splitPath)
```

Comment:
ditto

Reply:
We can add a new shim class.

Reply:
Is adding a shim to distinguish between 2.3 and 2.3+?

Reply:
yes, <2.3 and 2.3+